### ISSUE TITLE
inverse vector correction not working anymore

### ISSUE THREAD

**Issue by jk-rieck** [brodeau/sosie#17]
The inverse vector correction does not seem to work anymore in the most recent commits... I see, that the `-x` and `-y` switches have been remove in commit 4de70741b7e7235b32584c16c69391c1441f4be9 aready (I was using an earlier version before), however when eliminating `-x` and `-y` from my command
```shell
old:
corr_vect.x -I -i gird_U.nc grid_V.nc -t time_counter -x vozocrtx -y vomecrty -m mask.nc
new:
corr_vect.x -I -i gird_U.nc grid_V.nc -t time_counter -m mask.nc
``` 
I just get the "Usage"-message as if I typed `corr_vect.x -h`. Is the inverse correction not supported anymore? Or do I need to give specific names to the variables for this to work?
Any help would be much appreciated!

**Comment by HandmannP** [comment]
Problem is still there - did you already found a solution to this problem?

**Comment by mscheinert** [comment]
It seems, the *corr_vect* routine needs a **namelist**.

I guess, the namelist argument `-f namelist` is just missing here? Furthermore: the routine expects actually two namelists on disk, one with suffix "_x" the other with "_y", i.e. `namelist_x` and `namelist_y`.

**Comment by brodeau** [comment]
Hello all,
Yes it was in bad shape...
I made the usage message of "corr_vect.f90" clearer so it might now be understood that just pointed out buy Markus, in normal rotation mode `corr_vect.f90` requites two namelists: `<common-prefix>_x` and `<common-prefix>_y` (`<common-prefix>_x` being specified with the `-f` switch)...
I also fixed the inverse rotation capability (-I) of "corr_vect.f90" and added the new test script `examples/vector_correction/test_inverse_corr_vect_ORCA1.sh` so you guys have a working test case!
So basically the command for an inverse rotation looks like something like this:
`corr_vect.x -I -i fileU.nc fileV.nc -v u10 v10 -t time_counter -m mesh_mask_ORCA1v2_light.nc4`


Cheers, /laurent

**Comment by HandmannP** [comment]
Dear Laurent,

Thanks for the quick help! 
Now it runs through as did corr_vect.x with the sosie2 installation that I used previously.
But it does not save the unrotated files in the end even though it displays:

 Files created:
file1 
file2

Cheers,
Patricia

**Comment by brodeau** [comment]
Dear Patricia,

First, can you successfully perform the `test_corr_vect_ORCA1.sh` and `test_inverse_corr_vect_ORCA1.sh` found into `examples/vector_correction` ?
And then, does `test_inverse_corr_vect_ORCA1.sh` properly creates:
`u10_1x1-deg-ORCA1_gridT_grid_T_unrotated.nc`
`v10_1x1-deg-ORCA1_gridT_grid_T_unrotated.nc`
as expected?

/laurent

**Comment by HandmannP** [comment]
Hey Laurent, 

yes, the examples work - I will try to find the error on my side now - thanks for the quick help!

**Comment by brodeau** [comment]
What compiler do you use (brand and version)?
Any error messages?

**Comment by HandmannP** [comment]
There is no error message but I figured It would be good to first
transfer U and V both to the T grid and then correcting them, since they need to be on the same grid right?
Or is that irrelevant? I am looking for an error of some kind in my nameless files now.

**Comment by brodeau** [comment]
What are you trying to do and for what purpose, what model etc?

**Comment by HandmannP** [comment]
I am trying to transfer a high-res Nemo based tricolor model mean u and v field to 
a regular predefined field. 
So the vectors need to be corrected and interpolated to the new field. Hence I always did first the corr_vect and then sosie.

**Comment by brodeau** [comment]
it's a very "unusual need" to have to unrotate a vector hehe

**Comment by brodeau** [comment]
oh I see, ok makes sense then!

**Comment by HandmannP** [comment]
Yeah I know. So what is your opinion:
Is it better to first interpolate to the T grid then correct the vecors and then interpolate to the final grid or is first the correction and then interpolation the better version?

**Comment by brodeau** [comment]
Mhh, that's a good one...
Right now I don't remember what the "corr_vect" code expects as input... I will double check...
I would personnaly go for the interpolation U,V NEMO -> T NEMO before rotating, but need to know the answer to my previous point...

If you are really stuck with your bug, provide me somewhere on some cloud/server a tar file including:
* the U and V files (with only a couple of time records so it's not a huge download!)
* coresponging mesh_mask file

**Comment by brodeau** [comment]
I just comited some small changes in the code...
Could you look at `src/corr_vect.f90` and search and pay attention to the lines that I flagged with `#LB` ??? 
That's exactly your problem and I don't know...

**Comment by HandmannP** [comment]
For me - it seems like U and V are on U/V grids - so It appears that one does not need to bring them both to the T grid before unrotating. But I am not an expert for fortran.
